staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmapleaselock.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcelock
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ConfigMapLeaseLock is used for migration between the configmap and the lease
+type ConfigMapLeaseLock struct {
```

Comment:
can ConfigMapLeaseLock and EndpointsLeaseLock collapse to a `MultiLock` object that combines two lock interface implementations?

Reply:
Should we implement MultiLock only contains two locks, or uncertainly quantity locks?

Reply:
The usecase for two is migration. I don't see any usecase for more than two.
So I think we shouldn't overgeneralize - two is enough.